src/utils/generate_structs.jl

```diff
@@ -53,6 +53,16 @@ function {{struct_name}}{T}({{#parameters}}{{^internal}}{{name}}{{#default}}={{d
     {{struct_name}}({{#parameters}}{{^internal}}{{name}}, {{/internal}}{{/parameters}}InfrastructureSystemsInternal())
 end
 {{/parametric}}
+
+{{#defines_ext}}
+function {{struct_name}}({{#parameters}}{{^internal}}{{^ext}}{{^_forecasts}}{{name}}, {{/_forecasts}}{{/ext}}{{/internal}}{{/parameters}}; ext={{#parameters}}{{#ext}}{{default}}{{/ext}}{{/parameters}})
+    {{#parameters}}
+    {{/parameters}}
```

Comment:
Isn't this unnecessary?

Reply:
I will test it, I based it from the code in line 51.